src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -477,8 +473,19 @@ object ScalafixPlugin extends AutoPlugin {
       // used to signal that one of the argument cannot be reliably stamped
       object StampingImpossible extends RuntimeException with NoStackTrace
 
-      implicit val stamper = new CacheKeysStamper {
-        override protected def stamp: Arg.CacheKey => Unit = {
+      import sjsonnew._
+      import sjsonnew.BasicJsonProtocol._
+
+      val argWriter = new JsonWriter[Arg.CacheKey] {
+
+        override final def write[J](
+            obj: Arg.CacheKey,
+            builder: Builder[J]
+        ): Unit = stamp(obj)(builder)
+
+        private def stamp[J](
+            obj: Arg.CacheKey
+        )(implicit builder: Builder[J]): Unit = obj match {
```

Comment:
There is way too much logic in `ScalafixPlugin.scala`. I'll try to extract that in a separate PR.